sdk/nodejs/runtime/resource.ts

```diff
@@ -258,17 +260,35 @@ async function prepareResource(label: string, res: Resource, custom: boolean,
     // The list of all dependencies (implicit or explicit).
     const allDirectDependencies = new Set<Resource>(explicitDirectDependencies);
 
-    const allDirectDependencyURNs = await getResourceURNs(explicitDirectDependencies);
+    const allDirectDependencyURNs = await getCustomResourceURNs(explicitDirectDependencies);
```

Comment:
Please change the name of this local to indicate that it only includes custom resources. Likewise below.

Reply:
done.